### ISSUE TITLE
cannot build image: invalid reference format

### ISSUE THREAD

**Issue by chaunceyjiang** [kubernetes-sigs/kwok#182]
### How to use it?

- [ ] kwok
- [ ] kwokctl --runtime=docker (default runtime)
- [ ] kwokctl --runtime=binary
- [ ] kwokctl --runtime=nerdctl
- [ ] kwokctl --runtime=kind

### What happened?

<img width="1439" alt="image" src="https://user-images.githubusercontent.com/17962021/209645603-0dd6565a-8fbd-4551-b148-f3129981a671.png">

### What did you expect to happen?

can build image

### How can we reproduce it (as minimally and precisely as possible)?

1. git clone https://github.com/kubernetes-sigs/kwok.git
2. make build-image

### Anything else we need to know?

https://github.com/kubernetes-sigs/kwok/blob/a22f3cf7beeeafac7d8f792a900ba03fa23bd8a0/Makefile#L52-L55

If `STAGING_IMAGE_PREFIX` is not set, then `KWOK_IMAGE` and `CLUSTER_IMAGE` is `/kwok` and `/cluster`. This is the wrong format.

### OS version

<details>
```console        
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

Linux kk-instance-swjs2 3.10.0-1127.el7.x86_64 #1 SMP Tue Mar 31 23:36:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
# On Darwin:
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```
</details>

**Comment by chaunceyjiang** [comment]
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/17962021/209648212-e0de1661-7a0b-4513-90a2-149d61b079c9.png">

<img width="1433" alt="image" src="https://user-images.githubusercontent.com/17962021/209648259-83445aaa-df4c-4cd9-a2bf-9cbdebb01a8a.png">

**Comment by wzshiming** [comment]
https://github.com/kubernetes-sigs/kwok/blob/main/site/content/en/docs/contributing/development.md

You may refer to this document to compile

**Comment by chaunceyjiang** [comment]
Is `MAGE_PREFIX` required? Why not set a default value?

**Comment by chaunceyjiang** [comment]
I guess `IMAGE_ PREFIX` means docker registry, i think we can set a default value, like https://github.com/karmada-io/karmada/blob/master/Makefile#L6

**Comment by chaunceyjiang** [comment]
> https://github.com/kubernetes-sigs/kwok/blob/main/site/content/en/docs/contributing/development.md
> You may refer to this document to compile

still not working.

![image](https://user-images.githubusercontent.com/17962021/209747122-ddbfde07-2c47-4154-a364-d60563b60961.png)

@wzshiming

**Comment by wzshiming** [comment]
https://github.com/kubernetes-sigs/kwok/blob/86dd584aec6416a123fb1acc3e45613c5cb48d90/images/kwok/build.sh#L19

It seems that the relative path is not found when getting the file here.
Can you fill in the full log text?

**Comment by Fish-pro** [comment]
> > https://github.com/kubernetes-sigs/kwok/blob/main/site/content/en/docs/contributing/development.md
> > You may refer to this document to compile
> 
> still not working.
> 
> ![image](https://user-images.githubusercontent.com/17962021/209747122-ddbfde07-2c47-4154-a364-d60563b60961.png)
> 
> @wzshiming

@chaunceyjiang 
First execution
```sh
GOOS=linux make build
```
Then
```
IMAGE_PREFIX=registry.k8s.io/kwok make image
```

**Comment by chaunceyjiang** [comment]
<img width="804" alt="image" src="https://user-images.githubusercontent.com/17962021/209925705-f035846e-1d29-4377-9e51-fd8729a131a5.png">

`realpath` does not have this `--relative-to`  parameter.

**Comment by chaunceyjiang** [comment]
Can we use `readlink` instead of` realpath`?
like https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/run-e2e-tests.sh#L23

**Comment by wzshiming** [comment]
It seems that darwin's `realpath` is not compatible with linux, 
Maybe `brew install coreutiles` can avoid this.

Here is to get the relative path, if there is a better way to get the relative path that would be good.

**Comment by chaunceyjiang** [comment]
> if there is a better way to get the relative path that would be good.

why? 


<img width="1042" alt="image" src="https://user-images.githubusercontent.com/17962021/209951201-550cc7cb-67fb-47ec-950e-4ed861192166.png">

I think absolute paths are the recommended usage.

**Comment by wzshiming** [comment]
In the case of relative paths, the build log shows
```
docker buildx build --tag=kwok:test --platform=linux/amd64 --load -f images/kwok/Dockerfile .
```

when use absolute paths it doesn't look very clear

**Comment by wzshiming** [comment]
Fixed by #186

/close
